### PR TITLE
Full screen handler

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by ghsolomon** [comment]
closes #2924

**Comment by ghsolomon** [comment]
Implements DashCanvasView full-screen mode (#2959)

**Comment by lbwexler** [comment]
Very cool!  Would like to go over how we are packaging this.

**Comment by ghsolomon** [comment]
Trying to come up with the best name for this component.  Some ideas so far:

- FullScreen
- Expander
- FullScreenHandler
- LightBox

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+### 🎁 New Features
+
+* A new `FullScreenSupport` component for providing a child component with full screen expansion
+  capabilities
```
Comment (lbwexler) [inline CHANGELOG.md:6]: could sell this a bit harder.  also worth mentioning the enhancements to various hoist components (3!) to use it, allowing seamless, state-preserving, zoom capabilities

Comment (ghsolomon) [reply CHANGELOG.md:6]: Fair enough!

File: admin/tabs/activity/tracking/charts/ChartsModel.js [hunk 2]
```diff
@@ -7,15 +7,16 @@
 import {ChartModel} from '@xh/hoist/cmp/chart';
 import {br, fragment} from '@xh/hoist/cmp/layout';
 import {HoistModel, managed, lookup} from '@xh/hoist/core';
+import {FullScreenSupportModel} from '@xh/hoist/desktop/cmp/fullscreenhandler/FullScreenSupportModel';
```
Comment (lbwexler) [inline admin/tabs/activity/tracking/charts/ChartsModel.js:10]: directory name obsolete -- obviously we can update when we choose name

**Comment by lbwexler** [comment]
closing this in favor of https://github.com/xh/hoist-react/pull/3019

**Pull request closed without merging**